go/teams/handler.go

```diff
@@ -456,3 +461,27 @@ func handleSeitanSingle(ctx context.Context, g *libkb.GlobalContext, team *Team,
 
 	return nil
 }
+
+func handleSeitanSingleV2(key keybase1.SeitanPubKey, invite keybase1.TeamInvite, seitan keybase1.TeamSeitanRequest) (err error) {
+	pubKey, err := ImportSeitanPubKey(key)
+	if err != nil {
+		return err
+	}
+
+	var sig SeitanSig
+	decodedSig, err := base64.StdEncoding.DecodeString(string(seitan.Akey)) // For V2 the server responds with sig in the akey field
+	copy(sig[:], decodedSig[:])
+
+	now := keybase1.Time(seitan.UnixCTime) // For V2 this is ms since the epoch, not seconds
+	msg, err := GenerateSeitanSignatureMessage(seitan.Uid, seitan.EldestSeqno, SCTeamInviteID(seitan.InviteID), now)
```

Comment:
Regenerating the signed payload here can cause compatibility issues if we ever want to make changes that "should've been backwards compatible". For example, say in the future we add a new flag called `LaunchConfetti bool`. We might be inclined to think:

> Hey, old clients won't know to launch confetti here when an invite sets this to `true`, but that's kind of an optional feature anyway, so we're just gonna add this field without bumping the major seitan version.

That's _almost_ true, except that the old clients also won't know to reserialize the `LaunchConfetti` field here in this line, and the signature verification below will fail because the resulting bytes are different. This has caused problems for us in the past, with reverse signatures in sibkey links.

Anyway, it sounds like @maxtaco already thought about it a little bit, and we think it's not worth worrying about here. Seems fine to me, since there's a time limit on this data anyway, so our hands aren't tied as tightly as they are with sigchains. Maybe le's just leave a comment here to this effect?

Another variant of the same issue is that we need to make sure we're always serializing fields in the same _order_. It seems like we are, or else the tests wouldn't be passing?

Reply:
Tests cover the ordering case, but agreed we should make a note!